src/coreclr/dlls/mscoree/exports.cpp

```diff
@@ -221,7 +235,12 @@ int coreclr_initialize(
 
     g_hostpolicy_embedded = hostPolicyEmbedded;
 
-    if (pinvokeOverride != nullptr)
+    if (hostContract != nullptr)
+    {
+        HostInformation::SetContract(hostContract);
+    }
+
+    if (pinvokeOverride != nullptr && (hostContract == nullptr || hostContract->pinvoke_override == nullptr))
```

Comment:
I know this is a draft, so maybe work in progress, but this looks wrong. It will only use (and set) it if the hostcontract didn't specify the property directly. I would expect this to work the same as bundle probe (meaning it can be set by either a property or contract).

Reply:
Ah, yeah. I originally had `HostInformation::SetContract` setting everything else, but it was weird (especially for the bundle probe), so I moved it out, but missed going back to move the pinvoke override.